srcjs/init_shiny.js

```diff
@@ -288,6 +288,89 @@ function initShiny() {
       initialValues['.clientdata_output_' + id + '_height'] = this.offsetHeight;
     }
   });
+
+  function getComputedBgColor(el) {
+    if (!el) {
+      // Top of document, can't recurse further
+      return null;
+    }
+
+    let props = window.getComputedStyle(el);
+    let bgColor = props.getPropertyValue("background-color");
+    let m = bgColor.match(/^rgba\(\s*([\d.]+)\s*,\s*([\d.]+)\s*,\s*([\d.]+)\s*,\s*([\d.]+)\s*\)$/);
+    if (bgColor === "transparent" || (m && parseFloat(m[4]) === 0)) {
+      // No background color on this element. See if it has a background image.
+      let bgImage = props.getPropertyValue("background-image");
+      if (bgImage && bgImage !== "none") {
+        // Failed to detect background color, since it has a background image
+        return null;
+      } else {
+        // Recurse
+        return getComputedBgColor(el.parentElement);
+      }
+    }
+    return bgColor;
+  }
+
+  // Compute the color property of an a tag, scoped within the element
+  function getComputedLinkColor(el) {
+    let a = document.createElement("a");
+    a.href = "/";
+    let div = document.createElement("div");
+    div.style.setProperty("position", "absolute", "important");
+    div.style.setProperty("top", "-1000px", "important");
+    div.style.setProperty("left", "0", "important");
+    div.style.setProperty("width", "30px", "important");
+    div.style.setProperty("height", "10px", "important");
+    div.appendChild(a);
+    el.appendChild(div);
+    let linkColor = window.getComputedStyle(a).getPropertyValue("color");
+    el.removeChild(div);
+    return linkColor;
+  }
+
+  function getComputedFont(el) {
+    let style = window.getComputedStyle(el);
+    let fontFamily = style.getPropertyValue("font-family");
+    let fontSize = style.getPropertyValue("font-size");
+    let font = {
+      families: fontFamily.replace(/"/g, '').split(", "),
+      size: fontSize
+    };
+
+    // If the FontFaceSet API is available, use it to determine
+    // which font-family is supported by the browser
+    // https://developer.mozilla.org/en-US/docs/Web/API/FontFaceSet/check
+    // https://caniuse.com/#feat=mdn-api_fontfaceset_check
+    if (!document.fonts) {
+      return font;
+    }
+
+    for (let i = 0; i < font.families.length; i++)  {
+      let family = font.families[i];
+      let hasFont = false;
+      try {
+        hasFont = hasFont || document.fonts.check(font.size + " " + family);
+      } catch(err) {
+
+      }
+      if (hasFont) {
+        font.renderedFamily = family;
+        break;
+      }
+    }
+
+    return font;
+  }
+
+  $('.shiny-image-output, .shiny-plot-output, .shiny-report-theme').each(function() {
+    var id = getIdFromEl(this);
+    initialValues['.clientdata_output_' + id + '_bg'] = getComputedBgColor(this);
+    initialValues['.clientdata_output_' + id + '_fg'] = window.getComputedStyle(this).getPropertyValue("color");
```

Comment:
Could compute style once and pass to `getComputedFont()` and `getComputedBgColor()`

Reply:
We did some testing and `getComputedStyle()` appears to be very fast...on the order of 30-40ms for 10,000 calls to it